services/src/main/java/org/exoplatform/news/rest/NewsTargetingRestResourcesV1.java

```diff
@@ -53,8 +54,9 @@ public NewsTargetingRestResourcesV1(NewsTargetingService newsTargetingService) {
   @ApiResponses(value = { @ApiResponse(code = HTTPStatus.OK, message = "Request fulfilled"),
       @ApiResponse(code = HTTPStatus.INTERNAL_ERROR, message = "Internal server error"), })
   public Response getTargets(@Context HttpServletRequest request) {
+    org.exoplatform.services.security.Identity currentIdentity = ConversationState.getCurrent().getIdentity();
     try {
-      List<NewsTargetingEntity> targets = newsTargetingService.getTargets();
+      List<NewsTargetingEntity> targets = newsTargetingService.getTargets(currentIdentity);
       return Response.ok(targets).build();
     } catch (Exception e) {
```

Comment:
you need to catch IllegalAccessException and return Response.Status.UNAUTHORIZED, see NewsRestResourcesV1 methods